### DISCUSSION TITLE
Fine-Tuning with Tacotron2-DDC and GlowTTS

### DISCUSSION THREAD

**Discussion by meliksahturker** [coqui-ai/TTS#1147] [General Q&A]
I am tuning single speaker TTS models namely, "tts_models--en--ljspeech--glow-tts" and "tts_models--en--ljspeech--tacotron2-DDC" on my 2 hours of parallel text-audio (1085 wav files) which is formatted according to [Formatting Your Dataset guide](https://tts.readthedocs.io/en/latest/formatting_your_dataset.html). 

I am using CLI, which uses train_glowtts.py and train_tacotron_ddc.py under recipes/ljspeech/**/.

#### glow-tts
CUDA_VISIBLE_DEVICES="0" python recipes/ljspeech/glow_tts/train_glowtts.py --restore_path  /home/ubuntu/.local/share/tts/tts_models--en--ljspeech--glow-tts/model_file.pth.tar --coqpit.run_name "glow-tts-finetune-gulshatv2" --coqpit.lr 0.00001 --coqpit.batch_size 64 --coqpit.epochs 10

#### tacotron2
CUDA_VISIBLE_DEVICES="0" python recipes/ljspeech/tacotron2-DDC/train_tacotron_ddc.py --restore_path  /home/ubuntu/.local/share/tts/tts_models--en--ljspeech--tacotron2-DDC/model_file.pth.tar --coqpit.run_name "tacotron2-ddc" --coqpit.lr 0.00001 --coqpit.batch_size 16 --coqpit.epochs 10

I have issues with both models:

- Tacotron2-DDC: Despite trying different number of epochs, it doesn't save any models when training ends. I checked output_path in the training script, its the path of the py file but it just does not save anything other than training logs. I tried setting output path explicitly via argument: --output_path tuned_tacotron2-ddc.tar with no resolution.
Below are config and trainer log files:
[config](https://drive.google.com/file/d/19Pf-HWBmDe1wGm9FVOIuTuB08dV_Yg3Q/view?usp=sharing)
[train_log](https://drive.google.com/file/d/1MjmUy5dtq98gkFMo4Q5xa6BaB2ct2hCI/view?usp=sharing)

- GlowTTS: During inference, fine-tuned model produces 1 second of random-noise audio. I tried different texts, too. I thought I might have fine-tuned for too long or used a large learning_rate. So I fine-tuned with a learning rate of 0.000001 and 1 epoch, with the idea the original model would remain almost the same. Inference on this model also results in random-noise audio.
[config](https://drive.google.com/file/d/1oUtG_nTckc-6B70GKcXDlnXc5Jp2XlI2/view?usp=sharing)
[train_log](https://drive.google.com/file/d/1Fy4qa1NnGcMaAvhZUaEgPQdaxg51agck/view?usp=sharing)
[generated_audio](https://drive.google.com/file/d/1r2R_KoOju9ZrMTwq1pkORjmDVch9lHVP/view?usp=sharing)

What am I missing?

Edit: I was able to obtain checkpoint for tacotron2-ddc using train_tacotron_ddc.py directly. However, despite setting lr= 1e-8, epochs = 1 and changing dataset to a subset of original LJSpeech dataset (speaker 1, first 186 tracks) the inference produces static noise.

**Comment by meliksahturker** [comment]
Short update in issue: Fine-tuning according to [here](https://tts.readthedocs.io/en/latest/finetuning.html) neither creates a checkpoint nor a model output.

**Reply by Ca-ressemble-a-du-fake** [reply]
@meliksahturker I used the same tutorial that you referenced and the checkpoints are output in the outputpath on my Colab notebook. Maybe you used the same path for outputpath as for where you run the recipe from (I used a model dir for outputpath and a config dir for the recipe location).

**Reply by meliksahturker** [reply]
Thank you for your response. I don't know what makes the difference but following the same procedure, I have been able to fine-tune VITS model, with no issue with checkpoint or inference.
